src/Nethermind/Nethermind.Trie/Pruning/OverlayTrieStore.cs

```diff
@@ -12,10 +13,14 @@ public class OverlayTrieStore(IKeyValueStoreWithBatching? keyValueStore, IReadOn
     public override bool IsPersisted(Hash256? address, in TreePath path, in ValueHash256 keccak) =>
         base.IsPersisted(address, in path, in keccak) || store.IsPersisted(address, in path, in keccak);
 
-    public override TrieNode FindCachedOrUnknown(Hash256? address, in TreePath path, Hash256? hash)
+    internal override TrieNode FindCachedOrUnknown(Hash256? address, in TreePath path, Hash256? hash, bool isReadOnly)
     {
-        TrieNode node = base.FindCachedOrUnknown(address, in path, hash);
-        return node.NodeType == NodeType.Unknown ? store.FindCachedOrUnknown(address, in path, hash) : node;
+        ArgumentNullException.ThrowIfNull(hash);
+
+        TrieNode node = base.FindCachedOrUnknown(address, in path, hash, isReadOnly);
+        return node.NodeType == NodeType.Unknown
+            ? store.FindCachedOrUnknown(address, in path, hash) // no need to pass isReadOnly - IReadOnlyTrieStore overrides it as true
```

Comment:
Maybe we should pass it for completeness?

Reply:
Method with additional `isReadOnly` parameter is internal and not present in `IReadOnlyTrieStore` interface. This kind of makes sense, so would prefer to not add it.